EventFilter/HcalRawToDigi/plugins/HcalDigiToRawuHTR.cc

```diff
@@ -0,0 +1,297 @@
+  // loop over each uHTR and format data
+  // -----------------------------------------------------
+  // -----------------------------------------------------
+  // loop over each uHTR and format data
+  int idxuhtr =-1;
+  for( UHTRMap::iterator uhtr = uhtrs.uhtrs.begin() ; uhtr != uhtrs.uhtrs.end() ; ++uhtr){
+
+    idxuhtr ++;
+   
+    uint64_t crateId = (uhtr->first)&0xFF;
+    uint64_t slotId =  (uhtr->first&0xF00)>>8;
+
+    uhtrs.finalizeHeadTail(&(uhtr->second),_verbosity);
+    int fedId = FEDNumbering::MINHCALuTCAFEDID + crateId;
+    if( fedMap.find(fedId) == fedMap.end() ){
+      /* QUESTION: where should the orbit number come from? */
+      fedMap[fedId] = new HCalFED(fedId,iEvent.id().event(),iEvent.orbitNumber(),iEvent.bunchCrossing());
+    }
+    fedMap[fedId]->addUHTR(uhtr->second,crateId,slotId);
+  }// end loop over uhtr containers
+
+  /* ------------------------------------------------------
+     ------------------------------------------------------
+           putting together the FEDRawDataCollection
+     ------------------------------------------------------
+     ------------------------------------------------------ */
+  for( map<int,HCalFED*>::iterator fed = fedMap.begin() ; fed != fedMap.end() ; ++fed ){
+
+    int fedId = fed->first;
+
+    FEDRawData* rawData =  fed->second->formatFEDdata(); // fed->second.fedData;
+    FEDRawData& fedRawData = fed_buffers->FEDData(fedId);
+    fedRawData = *rawData;
+    delete rawData;
+
+    FEDHeader hcalFEDHeader(fedRawData.data());
+//    hcalFEDHeader.set(fedRawData.data(), 1, iEvent.id().event(), 1, fedId);
+    hcalFEDHeader.set(fedRawData.data(), 1, iEvent.id().event(), iEvent.bunchCrossing(), fedId);
+    FEDTrailer hcalFEDTrailer(fedRawData.data()+(fedRawData.size()-8));
+    hcalFEDTrailer.set(fedRawData.data()+(fedRawData.size()-8), fedRawData.size()/8, evf::compute_crc(fedRawData.data(),fedRawData.size()), 0, 0);
+
+  }// end loop over FEDs with data
+
+  iEvent.put(fed_buffers);
+  
+}
+
+
+// ------------ method called once each job just before starting event loop  ------------
+void HcalDigiToRawuHTR::beginJob(){}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void HcalDigiToRawuHTR::endJob(){}
+
+// ------------ method called when starting to processes a run  ------------
+void HcalDigiToRawuHTR::beginRun(edm::Run const&, edm::EventSetup const&){}
+
+// ------------ method called when ending the processing of a run  ------------
+void HcalDigiToRawuHTR::endRun(edm::Run const&, edm::EventSetup const&){}
+
+// ------------ method called when starting to processes a luminosity block  ------------
+void HcalDigiToRawuHTR::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&){}
+
+// ------------ method called when ending the processing of a luminosity block  ------------
+void HcalDigiToRawuHTR::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&){}
+
```

Comment:
please remove all emtpy overloads. They are not needed until actually overloaded.